text/text.go

```diff
@@ -300,3 +302,51 @@ func Draw(dst *ebiten.Image, text string, face font.Face, x, y int, clr color.Co
 
 	textM.Unlock()
 }
+
+// MeasureString measures the size of a given string using a given font.
```

Comment:
Rather than returning sizes, what about returning bounds?

Reply:
With "bounds", do you mean Go's image.Rectangle? I intentionally didn't use that because it comes with X and Y coordinates that would be left unused since they'd always be (0, 0).

Reply:
>  I intentionally didn't use that because it comes with X and Y coordinates that would be left unused since they'd always be (0, 0).

(X, Y) can be a relative position when assuming (0, 0) as a dot position, like what BoundString does.

Reply:
In 78a34f9, I made MeasureString return two `image.Point`s, one for the origin and one for the bounds. I tried using an `image.Rectangle` instead, but since it'd almost always have a negative Y coordinate, retrieving the height would be really unintuitive.
This way, one can simply do `origin, bounds := text.MeasureString(str, face)` and get everything.

Reply:
As of 78a34f9, a DrawTextInBox method could look like this:
```go
func DrawTextInBox(dst *ebiten.Image, str string, face font.Face, x, y int) image.Point {
	origin, bounds := text.MeasureString(str, face)

	xf, yf := float64(x + origin.X), float64(y + origin.Y)
	wf, hf := float64(bounds.X), float64(bounds.Y)

	ebitenutil.DrawRect(dst, xf, yf, wf, hf, color.Black)
	text.Draw(dst, str, face, x, y, color.White)

	return bounds
}
```

Reply:
> since it'd almost always have a negative Y coordinate, retrieving the height would be really unintuitive.

I am not sure how unintuitive it would be: `(b.Max.Y - b.Min.Y)` should work.

1. To be exact, my suggestion is that the returned value would be [Rectangle26_6](https://pkg.go.dev/golang.org/x/image/math/fixed?tab=doc#Rectangle26_6)

2. You can calculate a float64 value from Int26_6 like this

```
func toFloat(x fixed.Int26_6) float64 {
    return float64(x >> 6) + float64(x & ((1 << 6) - 1)) / float64(1 << 6)
}
```

2. With the rectangle, the code would be like this:

```
func DrawTextInBox(dst *ebiten.Image, str string, face font.Face, x, y int) image.Point {
	bounds := text.MeasureString(str, face)

	xf, yf := float64(x) + toFloat(bounds.Min.X), float64(y) + toFloat(bounds.Min.Y)
	wf, hf := toFloat(bounds.Max.X - bounds.Min.X), toFloat(bounds.Max.X - bounds.Min.X) 

	// ...

	return bounds
}
```

Reply:
IMO it would unnecessarily bloat code for a neglegible increase in precision.
Int26_6 is unintuitive in that it's impossible to just cast it to another type, requiring bit fiddling or rounding to be used in other contexts.
Seeing as this would be an API for game devs who want something they can work with instead of font creators who might want the return value to have the same type as the freetype package, I'd rather stick with normal integers.

Reply:
In b03ea20, I changed MeasureString so that it uses fixed.Int26_6 internally and only rounds at the end. This should avoid accumulation of rounding errors for very long strings.

I also took the liberty of changing the fixed26_6ToFloat64 function to mirror your toFloat method, since there's little point in having two functions to almost exactly the same thing.

Reply:
> Seeing as this would be an API for game devs who want something they can work with instead of font creators who might want the return value to have the same type that's used throughout the freetype package, I'd rather stick with normal integers.

Aha, I understand your points. OK let's avoid returning fixed26_6.

1. Sorry for going back and forth, but could we have this function to return only the size? If we want the dot position, we can calculate it by ourselves. We could have another function, but let's consider this later.
2. Instead of `Round`, let's use `Ceil` so that the region would be safer to cover the text. (Or, retuning float64 might be another good option. As long as text.Draw takes integers, I'm fine with ints though.)

> I also took the liberty of changing the fixed26_6ToFloat64 function to mirror your toFloat method,

Oh I didn't realize that. Thank you for taking care of this!

Reply:
> Sorry for going back and forth

No problem! Better to be safe than sorry 😄

> ...could we have this function to return only the size? If we want the dot position, we can calculate it by ourselves.

To be able to locate a piece of drawn text in top-left-origin space, you need the font's height so you can subtract it from the text's starting dot position. This necessitates a call to `face.Metrics()`, which does [a bunch of stuff](https://github.com/golang/freetype/blob/master/truetype/face.go#L254) and afaik isn't cached anywhere. 
Since MeasureString already calls `face.Metrics()` as part of its normal operation, I'd say it's actually helpful to devs to do that work for them and return a ready-to-use origin point along with a size. It's probably a form of premature optimization, yes, but it saves devs a function call and keeps their code clean.

> Instead of Round, let's use Ceil

Good call. I pushed this change in 5ce4ba9.

Reply:
> Since MeasureString already calls face.Metrics() as part of its normal operation, I'd say it's actually helpful to devs to do that work for them and return a ready-to-use origin point along with a size. It's probably a form of premature optimization, yes, but it saves devs a function call and keeps their code clean.

I understand the performance concern, but calculating `Metrics` should not so heavy task. Then this sounds like, as you indicated, premature optimization. In the users' perspective, clean API is preferable to clean implementation. So I'd want to add this as another function if necessary.

Reply:
I removed the origin point in 4aca1b0 and edited the function's documentation comment a bit.